lib/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.register = (server, options, next) => {
+
+  server.ext('onPreResponse', (request, reply) => {
+
+    const settings = request.route.settings.plugins.totalCount;
+    const credentials = request.auth.credentials;
+    const query = request.query;
+    const filter = request.query.filter || {};
+
+    if (settings && credentials && query.include && query.include.indexOf('total_count') !== -1 && request.response.source) {
+
+      Bluebird.resolve(settings.model)
+      .then((Model) => {
+        if (typeof new Model().filter === 'function') {
+          return new Model().filter(filter, credentials).count('*');
+        } else {
+          return new Model().count('*');
+        }
+      })
+      .then((count) => {
+        request.response.source.total_count = parseInt(count);
+        reply.continue();
+      })
+      .catch((err) => reply(err));
+
+    } else {
+      reply.continue();
+    }
```

Comment:
Optional: invert the `if` so that it early returns via `reply.continue`. Then the rest of the code can be moved outside of the conditional.